File: testbed_ruby/test/associations/unidirectional_optional_one_test.rb
require 'test/unit'
require './src-gen-umple/student_a_i'
require './src-gen-umple/mentor_a_i'
require './src-gen-umple/program_a_i'

module CruiseTest
class UnidirectionalOptionalOneTest < Test::Unit::TestCase

  def test_set_student
    s = StudentAI.new(1)
    m = MentorAI.new("a")
    assert_equal("a",m.get_name)
    assert_equal(nil,m.get_student)

    m.set_student(s)
    assert_equal(s,m.get_student)
    assert_equal(1,m.get_student.get_number)
  end
  
  def test_deleteLeavesStudentAlone
    m = MentorAI.new("a")
    s = StudentAI.new(1)
    m.set_student(s)
    p = ProgramAI.new
    s.set_program(p)
    m.delete
    assert_equal(nil,m.get_student)
    assert_equal(nil,m.get_program)
    assert_equal(p,s.get_program)
    
  end
end
end

File: testbed_php/test/associations/UnidirectionalOptionalOneTest.php
<?php

class UnidirectionalOptionalOneTest extends UnitTestCase
{


  public function test_setStudent()
  {
    $s = new StudentAI(1);
    $m = new MentorAI("a");
    $this->assertEqual("a",$m->getName());
    $this->assertEqual(null,$m->getStudent());

    $m->setStudent($s);
    $this->assertEqual($s,$m->getStudent());
    $this->assertEqual(1,$m->getStudent()->getNumber());
  }
  

  public function test_deleteLeavesStudentAlone()
  {
    $m = new MentorAI("a");
    $s = new StudentAI(1);
    $m->setStudent($s);
    $p = new ProgramAI();
    $s->setProgram($p);
    $m->delete();
    $this->assertEqual(null,$m->getStudent());
    $this->assertEqual(null,$m->getProgram());
    $this->assertEqual($p,$s->getProgram());
    
  }
  
}
